implementations/rust/ockam/ockam_api/src/nodes/service/flow_controls.rs

```diff
@@ -13,16 +13,40 @@ impl NodeManagerWorker {
         ctx: &Context,
         req: &Request<'_>,
         dec: &mut Decoder<'_>,
-    ) -> Result<ResponseBuilder> {
-        let request: AddConsumer = dec.decode()?;
+    ) -> Result<ResponseBuilder, ResponseBuilder<Error>> {
+        let request: AddConsumer = match dec.decode() {
+            Ok(r) => r,
+            Err(e) => {
+                let err_body =
+                    Error::new(req.path()).with_message(format!("Unable to decode request: {}", e));
+                return Err(Response::bad_request(req.id()).body(err_body));
+            }
+        };
 
         let mut route = match local_multiaddr_to_route(request.address()) {
-            None => return Ok(Response::bad_request(req.id())),
+            None => {
+                let err_body = Error::new(req.path())
+                    .with_message(format!("Invalid address: {}", request.address()));
+                return Err(Response::bad_request(req.id()).body(err_body));
```

Comment:
IIUC, it doesn't matter now whether we return an Ok(Response) or an Err(Reponse) because they are both unwrapped when calling `encode_request_result` right?

Would it be the same if we just refactor this unwrapping? https://github.com/build-trust/ockam/blob/9dd38a7c399ba28664bec13b0340d44eb0891b0a/implementations/rust/ockam/ockam_api/src/nodes/service.rs#L909

Reply:
That would be ideal if everything returned a `ResponseBuilder<T>` within the services, however there's a large handful of services being called here that only return a `Vec<u8>`

I don't think it would be the right step to change those services either, though this was something that I debated.

Reply:
Ok, got it, good to know for the next iterations!